File: backend/app/api/owners/getAllOwners/GetAllOwnersController.java
package api.owners.getAllOwners;

import play.libs.Json;
import play.libs.concurrent.HttpExecutionContext;
import play.mvc.Controller;
import play.mvc.Result;
import play.mvc.Results;

import javax.inject.Inject;
import java.util.concurrent.CompletionStage;

public class GetAllOwnersController extends Controller {

    @Inject
    GetAllOwnersService ownerService;

    @Inject
    HttpExecutionContext ec;


    public CompletionStage<Result> getOwners() {
        // returns a Collection<Owner>
        return ownerService.getOwners()
                .thenApplyAsync(owners -> Json.toJson(owners), ec.current())
                .thenApply(json -> Results.ok(json));
    }

}

File: backend/modules/petclinic/app/petclinic/api/owners/getAllOwners/GetAllOwnersController.scala
package petclinic.api.owners.getAllOwners

import javax.inject.Inject
import play.api.libs.json.Json
import play.api.mvc.{AbstractController, ControllerComponents, Results}

class GetAllOwnersController @Inject()(service: GetAllOwnersService, cc: ControllerComponents) extends AbstractController(cc) with JsonSupport {
  implicit val ec = cc.executionContext


  def getOwners = cc.actionBuilder.async { request =>
    service
      .getOwners()
      .map(o => Results.Ok(Json.toJson(o)))
      .recover {
        case x => Results.BadRequest
      }
  }

}